windows/RNLocalize/RNLocalizeModule.cpp

```diff
@@ -0,0 +1,190 @@
+{
+    return locale.substr(0, locale.find("-"));
+}
+
+bool RNLocalizeModule::getIsRTL(std::string languageCode)
+{
+    return std::find(USES_RTL_LAYOUT.begin(), USES_RTL_LAYOUT.end(), languageCode) != USES_RTL_LAYOUT.end();
+}
+
+bool RNLocalizeModule::getUsesMetricSystem(std::string countryCode)
+{
+    return std::find(USES_IMPERIAL.begin(), USES_IMPERIAL.end(), countryCode) == USES_IMPERIAL.end();
+}
+
+std::string RNLocalizeModule::getTemperatureUnit(std::string countryCode)
+{
+    return std::find(USES_FAHRENHEIT.begin(), USES_FAHRENHEIT.end(), countryCode) != USES_FAHRENHEIT.end() ? "fahrenheit" : "celsius";
+}
+
+bool RNLocalizeModule::getUses24HourClock()
+{
+    return !to_string(winrt::Windows::Globalization::Calendar().GetClock()).compare("24HourClock");
+}
+
+std::vector<std::string> RNLocalizeModule::getCurrencies()
+{
+    std::vector<std::string> currencies;
+    for (winrt::hstring currency : winrt::Windows::Globalization::GeographicRegion().CurrenciesInUse())
+    {
+        currencies.push_back(winrt::to_string(currency));
+    }
+    return currencies;
+}
+
+std::string RNLocalizeModule::getScriptCode(winrt::hstring locale)
+{
+    return winrt::to_string(winrt::Windows::Globalization::Language(locale).Script());
+}
+
+std::string RNLocalizeModule::getCalendar()
+{
+    std::string calendar = to_string(winrt::Windows::Globalization::Calendar().GetCalendarSystem());
+
+    // Window's Calendar system can return any of these values:
+    // "EastAsianLunisolarCalendar", "GregorianCalendar", "HebrewCalendar, "HijriCalendar",
+    // "JapaneseCalendar", "JulianCalendar", "KoreanCalendar", "PersianCalendar",
+    // "TaiwanCalendar", "ThaiBuddhistCalendar", "UmAlQuraCalendar"
+    // In all cases they end in "Calendar" and RNL doesn't expect this, so we must truncate it.
+    calendar = calendar.substr(0, calendar.size() - 8);
+
+    // Special edge case: RNL expects "buddhist" not "thaibuddhist"
+    // see https://github.com/zoontek/react-native-localize/blob/master/src/types.ts
+    if (calendar == "ThaiBuddhist")
+    {
+        return "buddhist";
+    }
+
+    std::transform(calendar.begin(), calendar.end(), calendar.begin(), ::tolower);
+    return calendar;
+}
+
+std::string RNLocalizeModule::getTimeZone()
+{
+    // Use GetTimeZoneInformation to get the hour bias from GMT and then create a formatted string.
+    TIME_ZONE_INFORMATION TimeZoneInfo;
+    GetTimeZoneInformation(&TimeZoneInfo);
+
+    int hourBias = -TimeZoneInfo.Bias / 60;
+    char timeZoneBuffer[10];
+    snprintf(
+        timeZoneBuffer,
+        sizeof(timeZoneBuffer),
+        "GMT%s%02d:00",
+        hourBias < 0 ? "-" : "+",
+        hourBias < 0 ? -hourBias : hourBias);
+    std::string timeZoneString(timeZoneBuffer);
+
+    return timeZoneString;
+}
+
+NumberFormatSettings RNLocalizeModule::getNumberFormatSettings(std::string locale)
+{
+    LPCWSTR locale_l = std::wstring(locale.begin(), locale.end()).c_str();
```

Comment:
don't use L types, just use PCWSTR or wchar_t*. (prefer std:: types when possible though)
(here and other places)

Reply:
Removed all places of windows macros